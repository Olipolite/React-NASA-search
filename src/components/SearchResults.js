import React from "react";
import PropTypes from "prop-types";
import "../styles/searchresults.css";

const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p>No results</p>
  } else {
    return (
      <div className="all-images">
        {results.map((image, index) => (
        <img className="card-image" src={image} alt="spaceImage" key={index} />
        ))}
      </div>
    )
  };
};

export default SearchResults;

SearchResults.propTypes = {
  results: PropTypes.array.isRequired
}
