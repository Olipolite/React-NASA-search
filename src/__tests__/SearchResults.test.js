import { render, } from '@testing-library/react';
import SearchResults from '../components/SearchResults';

describe("Search results", () => {
  let resultsMock = ["test1", "test2"];

    it("renders correctly", () => {

      const { asFragment } = render(<SearchResults results={resultsMock} />);
  
      expect(asFragment()).toMatchSnapshot();
    });
  });