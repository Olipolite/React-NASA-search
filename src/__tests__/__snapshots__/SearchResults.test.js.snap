// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search results renders correctly 1`] = `
<DocumentFragment>
  <div
    class="all-images"
  >
    <img
      alt="spaceImage"
      class="card-image"
      src="test1"
    />
    <img
      alt="spaceImage"
      class="card-image"
      src="test2"
    />
  </div>
</DocumentFragment>
`;
